src/components/fx/click.js

```diff
@@ -22,12 +22,29 @@ module.exports = function click(gd, evt, subplot) {
         hover(gd, evt, subplot, true);
     }
 
-    function emitClick() { gd.emit('plotly_click', {points: gd._hoverdata, event: evt}); }
+    function emitClick(data) { gd.emit('plotly_click', {points: data, event: evt}); }
 
-    if(gd._hoverdata && evt && evt.target) {
-        if(annotationsDone && annotationsDone.then) {
-            annotationsDone.then(emitClick);
-        } else emitClick();
+    // TODO(j): remove gd._hoverdata
+    var clickmode = gd._fullLayout.clickmode;
+    var data;
+    if(evt && evt.target) {
+        if(gd._hoverdata) {
+            data = gd._hoverdata;
+        } else if(clickmode.indexOf('anywhere') > -1) {
+            var xaxis = gd._fullLayout.xaxis;
+            var yaxis = gd._fullLayout.yaxis;
+            var l = gd._fullLayout.margin.l;
+            var t = gd._fullLayout.margin.t;
+            var x = xaxis.p2c(evt.pointerX - l);
+            var y = yaxis.p2c(evt.pointerY - t);
+            data = [{x: x, y: y}];
+        }
+        if(data) {
+            if(annotationsDone && annotationsDone.then) {
+                // TODO(j) add gd._hoverdata to emitClick here
+                annotationsDone.then(emitClick);
```

Comment:
How do I pass data here. Not too familiar with JavaScript syntax

Reply:
```suggestion
                annotationsDone.then(function() { emitClick(data); });
```